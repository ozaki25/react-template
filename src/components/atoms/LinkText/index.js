import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

function LinkText({ children, onClick }) {
  return <Button onClick={onClick}>{children}</Button>;
}

export const Button = styled.button`
  color: #666;
  font-size: 16px;
  text-decoration: underline;
`;

LinkText.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
};

LinkText.defaultProps = {
  onClick: null,
};

export default LinkText;
