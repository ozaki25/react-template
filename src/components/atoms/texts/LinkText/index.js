import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const LinkText = ({ children, onClick }) => <Button onClick={onClick}>{children}</Button>;

const Button = styled.button`
  color: #666;
  font-size: 16px;
  text-decoration: underline;
`;

LinkText.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
};

LinkText.defaultProps = {
  onClick: null,
};

export default LinkText;
