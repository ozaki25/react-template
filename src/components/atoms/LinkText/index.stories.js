import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import LinkText from '.';

const stories = storiesOf('atoms', module);

stories.add(
  'LinkText',
  withInfo('リンク用文字列')(() => (
    <LinkText onClick={action('clicked LinkText')}>{text('text', 'リンク')}</LinkText>
  )),
);
