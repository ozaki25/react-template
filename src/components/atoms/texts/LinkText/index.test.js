import React from 'react';
import { snapshot } from 'test/helpers';
import LinkText from '.';

const props = {
  onClick: jest.fn(),
};

snapshot('LinkText', <LinkText {...props}>リンク</LinkText>);
