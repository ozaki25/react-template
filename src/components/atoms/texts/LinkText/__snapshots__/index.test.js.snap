// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: LinkText 1`] = `
<styled.button
  onClick={[Function]}
>
  リンク
</styled.button>
`;

exports[`should match snapshot: LinkText/Button 1`] = `
.c0 {
  color: #666;
  font-size: 16px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<button
  className="c0"
>
  リンク
</button>
`;
